kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/OperationSupport.java

```diff
@@ -216,24 +212,33 @@ public URL getResourceURLForPatchOperation(URL resourceUrl, PatchContext patchCo
     }
     return resourceUrl;
   }
+  
+  protected <T> T correctNamespace(T item) {
+    if (!isResourceNamespaced() || this.context.isDefaultNamespace() || !(item instanceof HasMetadata)) {
+      return item;
+    }
+    String itemNs = KubernetesResourceUtil.getNamespace((HasMetadata)item);
+    
+    if (Utils.isNotNullOrEmpty(namespace) && Utils.isNotNullOrEmpty(itemNs) && !namespace.equals(itemNs)) {
+      item = Serialization.clone(item);
+      KubernetesResourceUtil.setNamespace((HasMetadata)item, namespace);
+    }
+    return item;
+  }
```

Comment:
Not sure if we have an explicit test for this, but full coverage and documentation of conditions should be provided since it's determinant for the Namespace resolution logic.

I'm not sure if this PR also includes the `default` namespace fallback logic discussed in some meetings.

Reply:
> Not sure if we have an explicit test for this, but full coverage and documentation of conditions should be provided since it's determinant for the Namespace resolution logic.

There's an addition to the migration guide https://github.com/fabric8io/kubernetes-client/pull/3858/files#diff-6cf6f0d1132c6d642456eda0b231e404d9b3743c0c0ff86346055c5f7c78c268R15

And https://github.com/fabric8io/kubernetes-client/blob/cfd501a75f96c8e02e2e554ecdfb4d3dd56f161c/kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/NamespacedItemTest.java highlighting the usage of the explicit namespace.

> I'm not sure if this PR also includes the default namespace fallback logic discussed in some meetings.

It does not, that issue is if the Config has no default namespace, and still in the 6.0.0 bucket.  It is a little related to the discussion about what should inNamespace(null) mean.

Reply:
> And https://github.com/fabric8io/kubernetes-client/blob/cfd501a75f96c8e02e2e554ecdfb4d3dd56f161c/kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/NamespacedItemTest.java highlighting the usage of the explicit namespace.

This is what I was looking for :+1: :smiley: